packages/keyring-controller/src/KeyringController.ts

```diff
@@ -723,39 +732,49 @@ export class KeyringController extends BaseController<
     }
   }
 
-  async unlockQRHardwareWalletAccount(index: number): Promise<void> {
+  async unlockQRHardwareWalletAccount(index: number): Promise<string> {
     const keyring = await this.getOrAddQRKeyring();
 
     keyring.setAccountToUnlock(index);
     const oldAccounts = await this.#keyring.getAccounts();
     await this.#keyring.addNewAccount(keyring);
     const newAccounts = await this.#keyring.getAccounts();
     this.updateIdentities(newAccounts);
+    let newHardwareWalletAddress = '';
     newAccounts.forEach((address: string) => {
       if (!oldAccounts.includes(address)) {
         if (this.setAccountLabel) {
           this.setAccountLabel(address, `${keyring.getName()} ${index}`);
         }
-        this.setSelectedAddress(address);
+        newHardwareWalletAddress = address;
```

Comment:
Similar to a previous case, this variable will get updated every time `!oldAccounts.includes(address)`.

In addition it doesn't seem we are testing the return value of this method anywhere.

Reply:
Same here.

What do you think about this refactor? [b65201a](https://github.com/MetaMask/core/pull/1095/commits/b65201a505b537138003f24ba2497d6f8f1bf84d)

I'll await until I get your input before developing the proper test

Reply:
Yeah that makes a lot more sense.